src/main/kotlin/features/inventory/PetFeatures.kt

```diff
@@ -19,22 +27,60 @@ object PetFeatures : FirmamentFeature {
 
 	object TConfig : ManagedConfig(identifier, Category.INVENTORY) {
 		val highlightEquippedPet by toggle("highlight-pet") { true }
+		var petOverlay by toggle("pet-overlay") { false }
+		val petOverlayHud by position("pet-overlay-hud", 80, 10) { Point(0.5, 1.0) }
 	}
 
 	val petMenuTitle = "Pets(?: \\([0-9]+/[0-9]+\\))?".toPattern()
+	var activePetData: HypixelPetInfo? = null
+	var petItemStack: ItemStack? = null
 
 	@Subscribe
 	fun onSlotRender(event: SlotRenderEvents.Before) {
 		if (!TConfig.highlightEquippedPet) return
 		val stack = event.slot.stack
 		if (stack.petData?.active == true)
 			petMenuTitle.useMatch(MC.screenName ?: return) {
-				event.context.drawGuiTexture(
-					event.slot.x, event.slot.y, 0, 16, 16,
-					Identifier.of("firmament:selected_pet_background")
-				)
-			}
+			activePetData = stack.petData
+			petItemStack = stack
+			event.context.drawGuiTexture(
+				event.slot.x, event.slot.y, 0, 16, 16,
+				Identifier.of("firmament:selected_pet_background")
+			)
+		}
 	}
 
+	@Subscribe
+	fun onRenderHud(it: HudRenderEvent) {
+		if (!TConfig.petOverlay) return
+		val itemStack = petItemStack ?: return
+		val petData = activePetData ?: return
+		val rarity = Rarity.fromString(petData.tier.name)
+		val rarityCode = Rarity.colourMap[rarity]?.code ?: "f"
+		val xp = petData.level
+		val petType = titleCase(petData.type)
+		val heldItem = petData.heldItem?.let { item -> "Held Item: ${titleCase(item)}" }
+
+		it.context.matrices.push()
+		TConfig.petOverlayHud.applyTransformations(it.context.matrices)
+
+		val lines: MutableList<String> = ArrayList<String>()
```

Comment:
Prefer Text over raw Strings. This might neccessitate some extra `Text.literal` calls.

```suggestion
		val lines = mutableListOf<Text>()
```